astroid/inference.py

```diff
@@ -260,16 +262,22 @@ def infer_call(self, context=None):
                 yield from callee.infer_call_result(caller=self, context=callcontext)
         except InferenceError:
             continue
```

Comment:
```suggestion
            yield util.Uninferable
            continue
```
I'm wondering why we do not yield anything in case of inference error. We're yielding something if call is uninferable itself line 252/254.

Reply:
That's what the `@decorators.raise_if_nothing_inferred` decorator is for I think.

If there is no yielded value it will become `Uninferable` or raise `InferenceError`.

Changing this could break a lot of test: if one `callee's` result is inferable and one isn't currently we get one yielded value without ambiguity. If we would add `Uninferable` to this because the second `callee's` result is uninferable we introduce a lot of ambiguity.
I can think of definitions behind `sys` guards. We handle those pretty poorly so we might struggle to understand what a definition under such a guard returns.